fragments/filter/README.md

```diff
@@ -0,0 +1,632 @@
+
+API implementations advertise which `filter-lang` values are supported via conformance classes in the Landing Page.
+At least one must be implemented, but it is recommended to implement both. If both are advertised as conformance classes, the server should process either for a GET request, but may only process cql-json for a POST request. If POST of cql-text is not supported, the server must return a 400 error if `filter-lang=cql-text`.
+
+## Interaction with Endpoints
+
+Landing Page (`/`) returns:
+
+```json
+{
+  "id": "example_stacapi",
+  "conformsTo": [
+    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core",
+    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30",
+    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson",
+    
+    "http://www.opengis.net/spec/ogcapi_common-2/1.0/req/collections",
+
+    "http://www.opengis.net/spec/ogcapi-features-3/1.0/req/filter",
+    "http://www.opengis.net/spec/ogcapi-features-3/1.0/req/features-filter",
+    "http://www.opengis.net/spec/ogcapi-features-3/1.0/req/simple-cql",
+    "http://www.opengis.net/spec/ogcapi-features-3/1.0/req/cql-text",
+    "http://www.opengis.net/spec/ogcapi-features-3/1.0/req/cql-json",
+
+    "http://stacspec.org/spec/api/1.0.0-beta.1/core",
+    "http://stacspec.org/spec/api/1.0.0-beta.1/req/stac-search",
+    "http://stacspec.org/spec/api/1.0.0-beta.1/req/stac-response"
+  ],
+  "links": [
+    {
+      "title": "Search",
+      "href": "https://example.org/search",
+      "rel": "search",
+      "type": "application/geo+json"
+    },
+    {
+      "title": "Queryables",
+      "href": "https://example.org/queryables",
+      "rel": "http://www.opengis.net/def/rel/ogc/1.0/queryables",
+      "type": "application/schema+json"
+    }
+  ],
+  "stac_extensions": [],
+  "stac_version": "1.0.0",
+}
+```
+
+Client can use the link with `"rel": "http://www.opengis.net/def/rel/ogc/1.0/queryables"` to retrieve the queryables.
+
+Queryables endpoint (`/queryables`) returns:
+
+```json
+{
+  "$schema" : "https://json-schema.org/draft/2019-09/schema",
+  "$id" : "https://example.org/queryables",
+  "type" : "object",
+  "title" : "Queryables for Example STAC API",
+  "description" : "Queryable names for the example STAC API Item Search filter.",
+  "properties" : {
+    "id" : {
+      "description" : "ID",
+      "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json#/id"
+    },
+    "collection" : {
+      "description" : "Collection",
+      "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json#/collection"
+    },
+    "geometry" : {
+      "description" : "Geometry",
+      "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/item.json#/geometry"
+    },
+    "datetime" : {
+      "description" : "Datetime",
+      "$ref": "https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/datetime.json#/properties/datetime"
+    },
+    "eo:cloud_cover" : {
```

Comment:
This is one thing that's not clear to me, which may be worth explaining - if it's listed in queryables does that mean it's in all the data? Or can you put things in queryables that might be there? Like is the contract of it that the data will have this value? Or just that this is the name of the field to be filtered on, without a guarantee that it's there. If the latter then it could be interesting to just define the STAC API root queryables as the common metadata plus stable fields, and that API implementations can just put that up, without having to do any dynamic stuff.